.github/workflows/spellcheck.yml

```diff
@@ -1,28 +1,34 @@
-# This is a basic workflow to help you get started with Actions
+---
+# SPDX-license-identifier: Apache-2.0
+##############################################################################
+# Copyright (c) 2022
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Apache License, Version 2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+##############################################################################
 
 name: Spellcheck CI
-
-# Controls when the action will run.
+# yamllint disable-line rule:truthy
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
   pull_request:
-    branches: [ main ]
-
-  # Allows you to run this workflow manually from the Actions tab
+    types:
+      - submitted
+    paths:
+      - '**.md'
+      - '!.github/*'
+    branches:
+      - main
```

Comment:
Each localization team has (/will have) their own development branch (https://github.com/cncf/glossary/tree/dev-ko for Korean for instance). So, localization teams will not use `main` branch for localizing documents. We need to add development branch if a localization team wants to use this spellchecker in workflow.